library/Translate/src/Juvix/Contextify/Environment.hs

```diff
@@ -151,6 +126,59 @@ onExpression form trigger func =
     (trigger, func)
     (bindingForms, searchAndClosureNoCtx)
 
+------------------------------------------------------------
+-- Pass infrastructure
+------------------------------------------------------------
+
+data Pass m = Pass
+  { sumF :: SexpContext -> Sexp.Atom -> Sexp.T -> m Sexp.T,
+    termF :: SexpContext -> Sexp.Atom -> Sexp.T -> m Sexp.T,
+    tyF :: SexpContext -> Sexp.Atom -> Sexp.T -> m Sexp.T
+  }
+
+singlePass :: (SexpContext -> Sexp.Atom -> Sexp.T -> m Sexp.T) -> Pass m
+singlePass f = Pass f f f
+
+class
+  Context.ToContextFormGeneral contextTo =>
+  ToContextTraversal (pass :: (* -> *) -> *) contextTo
+    | pass -> contextTo
+  where
+  promotePass ::
+    (HasReader "closure" Closure.T m, HasThrow "error" ErrorS m) =>
+    pass m ->
```

Comment:
Should the `m` be part of the interface like MTL/Capabilities does it? Or should we just fix the effect in here?

Reply:
I think I like it where it is, as it's closest to where it's used, but it's not a strong feeling.